minetest/minetest
# Basic implementation of working server to server message.

- Goal of the PR
Add ability for server-to-server communication in connection with #14129.
- How does the PR work?
Send string from one to another server. It is on Lua to serialize/deserialize string and do something around it.
- Does it resolve any reported issue?
- Yes, it should fix part of #5813.
- If not a bug fix, why is this PR needed? What usecases does it solve?
With this, a network of servers should be created. So some networks like one server for space and another server for each planet, can be created and players can be transferred between servers by #14129 and their inventories and other states can be synced by server-to-server communication.

## To do

This PR is a Work in Progress.

- [ ] Apply feedback.
- [ ] Add better authorization.

## How to test

1. Compile Minetest with this branch.
1. Create two worlds with the game devtest.

1. To `server0` world directory save file [network_server_0.txt](https://github.com/minetest/minetest/files/13853410/network_server_0.txt) as `network.conf`.
1. To `server1` world directory save file [network_server_1.txt](https://github.com/minetest/minetest/files/13853411/network_server_1.txt) as `network.conf`.

1. Host server for game `server0` on port `30000`.
1. Host server for game `server1` on port `30000`.

1. Call command `/send_msg_to_server server0 HELLO_SERVER!` on `server1` or
   command `/send_msg_to_server server1 HELLO_SERVER!` on `server0`.
1. The first message is lost at the moment, so send the same message again to see it.
1. Received message should be printed as a message for all players on the second server.



https://github.com/minetest/minetest/assets/17455197/46de45f8-da97-4199-b79c-557c510ac4ba
